### ISSUE TITLE
Functions and IdentityMultiple 

### ISSUE THREAD

**Issue by ChristopherChudzicki** [mitodl/mitx-grading-library#102]
## The Issue
In `MatrixGrader` problems, if `IdentityMultiple` instances are passed to functions, students currently receive unhelpful generic error messages:

```python
from mitxgraders import FormulaGrader

grader = MatrixGrader(answers='1')
grader(None, 'sin(I)'
# raises:
# StudentFacingError: Invalid Input: Could not check input 'sin(I)'
```

This is bad.

(The actual cause of this error is an `isinf` check in calc.py's `eval_function` parse action. However: I suggest that this issue should be for discussing the intended behavior, not the source of the bug.)

## Original motivation of IdentityMultiple

My thinking with IdentityMultiple originally was:

- in MatrixGrader problems, 'I' for identity should always be available, so authors should not need to declare it as a specified constant.
- The dimension of the identity operator is usually irrelevant:
    - True for binary operations add/subtract and multiple/divide
    - True (**but not implemented**) for functions determinant, transpose, conjugation
    - False for tensor product; False for functions trace, norm
- In specific problems where identity's dimension does matter, the author should declare a`user_constant` with identity of relevant size. (Either overriding 'I', or declaring 'I_2', 'I_3', etc.)

## Resolutions

Shape is a bit more important for the identity matrix than I originally realized. In particular, I see **no way** to handle `trace` and `norm` for generic-dimensional identity operators.

I'm not sure what the best approach to resolve this is. Two ideas:

- **Idea A**: Implement functions like `det`, `trans`, etc, and provide special error messages for`trace` and`norm`.
    -  _Pros:_ Automatically declared `I` of generic dimension that works in almost all situations.
    - _Cons:_ Hard to maintain lots of special cases. Also, authors need to understand how IdentityMultiple works in order to implement their own matrix functions.

- **Idea B**: Abandon IdentityMultiple all together. Provide a helper function `Identity(n)` for authors to declare identity matrix of specific size through `user_constants`.
    - _Pros:_ More explicit; fewer special cases; automatically resolves #101; identities declared this way would work automatically with tensor products and any `numpy` matrix function. 
    - _Cons:_ Requires extra declaration by author. Requires author to think about dimensionality in problems where it might be irrelevant. 

I am personally rather fond of `IdentityMultiple`, but I am actually leaning toward Idea B.

**Comment by jolyonb** [comment]
Note that `det` is also dependent on the dimensionality.

I too am quite fond of `IdentityMultiple`, but I think you're right that Idea B is probably the better way to go. Being explicit about what something is will ultimately lead to less confusion than having a clever but ill-defined construct.

I propose that in addition to exposing a helper function `Identity(n)`, we have an additional key like `identity_dim` (default `None`) that will populate `I` with the identity of the appropriate dimension. This reduces the author load just that little bit.

**Comment by jolyonb** [comment]
Resolved by #105.
